tests/ui/attributes/link-dl.rs

```diff
@@ -0,0 +1,19 @@
+// Regression test for an issue discovered in https://github.com/rust-lang/rust/pull/143193/files and rediscovered in https://github.com/rust-lang/rust/issues/147254#event-20049906781
+// Malformed #[link] attribute was supposed to be deny-by-default report-in-deps FCW,
+// but accidentally was landed as a hard error.
+//
+// rev1 tests that with `ill_formed_attribute_input` (the default) denied,
+// the attribute produces an FCW
+// rev2 tests that with `ill_formed_attribute_input` allowed the test passes
+
+//@ revisions: rev1 rev2
+//@[rev2] check-pass
```

Comment:
Nit: can you call these revisions something more meaningful? Like `default_fcw` and `allowed` maybe?